Assets/Scripts/Layers/Adapters/DataTypeChain.cs

```diff
@@ -94,22 +92,28 @@ private IEnumerator DownloadAndCheckSupport(string url)
         /// <returns></returns>
         private IEnumerator DownloadDataToLocalCache(LocalFile urlAndData)
         {
-            var url = urlAndData.SourceUrl;
-            var uwr = UnityWebRequest.Get(url);
-            var optionalExtention = Path.GetExtension(url).Split("?")[0];
-            var guidFilename = Guid.NewGuid().ToString() + optionalExtention;
-            string path = Path.Combine(Application.persistentDataPath, guidFilename);
-
-            uwr.downloadHandler = new DownloadHandlerFile(path);
-            yield return uwr.SendWebRequest();
-            if (uwr.result == UnityWebRequest.Result.Success)
-            {
-                urlAndData.LocalFilePath = path;
-            }
-            else
+            var futureFileInfo = Resource<FileInfo>.At(urlAndData.SourceUrl).Value;
```

Comment:
In case anyone is wondering where the whole 'Guid' and path.Combine stuff went: the default behaviour of Uxios' Resource object -when using the type FileInfo as generic- is to do precisely this, generate a filename based on a GUID, put it in the persistentDataPath and if the original URL had an extension, it will include that.

This behaviour of Uxios was inspired by this method, as this is a really nice way to prevent name clashes; and by returning FileInfo, a framework can use this as an upload mechanism and use FileInfo's native methods to move a file to a definitive location if need. In this case: we don't need that